### PR TITLE
Improve user groups management

### PR DESCRIPTION
**Pull request by jraddaoui**
- Create "Viewers" group, only for display purposes.
- Allow sorting by groups in the users table:
  - Create groups concatenation sub-query to allow sorting by group.
  - Annotate query with field containing the sub-query result.
  - Use the annotated field to query, sort and display.

Connects to #11.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,6 +64,7 @@ The following steps are just an example of how to run the application in a produ
 
 * Python 3.4 or higher
 * Elasticsearch 6.x
+* MySQL or SQLite
```
Comment (sevein) [inline README.md]: My advise here is don't tell people to use MySQL until you have a build in CI verifying that it works properly. It's ok to use SQLite as long as it does not become a bottleneck.

File: accesspoc/dips/migrations/0010_add_viewers_user_group.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('dips', '0009_auto_20180721_1926'),
+    ]
+
+    operations = [
+        migrations.RunPython(add_viewers_user_group),
```
Comment (sevein) [inline accesspoc/dips/migrations/0010_add_viewers_user_group.py]: It's better if you provide up and down migrations. It's key for solid deployments to provide a rollback strategy. But if you're planning to squash the migrations before the first stable release then it's okay not to do it.

File: accesspoc/dips/views.py [hunk 3]
```diff
@@ -127,24 +127,32 @@ def users(request):
...
     if 'query' in request.GET and request.GET['query']:
         query = request.GET['query']
-        users = User.objects.order_by(sort_field).filter(
+        users = users.order_by(sort_field).filter(
             Q(username__icontains=query) |
             Q(first_name__icontains=query) |
             Q(last_name__icontains=query) |
             Q(email__icontains=query) |
-            Q(groups__name__icontains=query)
+            Q(group_names__icontains=query)
```
Comment (sevein) [inline accesspoc/dips/views.py]: I think that you should start considering moving this queries to the model layer - it doesn't need to happen in this PR. The views should be as thin as possible. You can do all sort of things in the model layer, e.g. see [proxies](https://docs.djangoproject.com/en/dev/topics/db/models/#proxy-models).

File: accesspoc/dips/migrations/0010_add_viewers_user_group.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+
+def add_viewers_user_group(apps, schema_editor):
+    """
+    Create 'Viewers' user group with no special permissions,
+    only for display purposes.
```
Comment (sevein) [inline accesspoc/dips/migrations/0010_add_viewers_user_group.py]: Wouldn't be useful to have a simple test to verify this behaviour?

**Comment by jraddaoui** [comment]
Thanks @sevein! I hope that we can squash the migrations before the beta release but I've addressed the other feedback.

**Pull request merged**